docs/user-guide/configure-certificates.md

```diff
@@ -1,30 +1,77 @@
+Creating a key ring and connecting the certificate key pair requires elevated permissions. When the TSO user ID does not have the authority to manipulate key rings and users want to create a Zowe sandbox environment or for testing purposes, the USS keystore is a good alternative.
+
+### PKCS12 certificates in a keystore
+
+Zowe is able to use PKCS12 certificates that are stored in USS. This certificate is used for encrypting TLS communication between Zowe clients and Zowe z/OS servers, as well as intra z/OS Zowe server to Zowe server communication. Zowe uses a `keystore` directory to contain its external certificate, and a `truststore` directory to hold the public keys of servers which Zowe communicates with (for example z/OSMF).
+
+By default, Zowe is reading PKCS12 keystore from `keystore` directory which can be located in zowe.yaml. This directory contains a server certificate, the Zowe generated certificate authority, and a `truststore` which holds intermediate certificates of servers that Zowe communicates with (for example z/OSMF).
+
+The use of a USS PKCS12 keystore is suitable for proof-of-concept projects as special permissions to create and manage the PKCS12 keystore are not required. For production usage of Zowe, it is recommended to work with certificates held in z/OS key rings. Configuring z/OS key rings may require security administrator privileges.  
+
+### JCERACFKS certificates in a key ring
+
+Zowe is able to work with certificates held in a **z/OS Key ring**.  
+
+The JCL member `.SZWESAMP(ZWEKRING)` contains security commands to create a SAF key ring. By default, this keyring is named `ZoweKeyring`. You can use the security commands in this JCL member to generate a Zowe certificate authority (CA) and sign the server certificate with this CA. The JCL contains commands for three z/OS security managers: RACF, TopSecret, and ACF/2.
+
+There are two ways to configure and submit `ZWEKRING`:
+
+- Customize and submit the `ZWEKRING` JCL member.
+- Customize the `zowe.setup.certificate` section in `zowe.yaml` and use the `zwe init certificate` command. 
+
+Use the `zwe init certificate` command to prepare a customized JCL member using `ZWEKRING` as a template.  
+
+A number of key ring scenarios are supported:
+
+- Creation of a local certificate authority (CA) which is used to sign a locally generated certificate. Both the CA and the certificate are placed in the `ZoweKeyring`.
+- Import of an existing certificate already held in z/OS to the `ZoweKeyring` for use by Zowe.  
+- Creation of a locally generated certificate and signed by an existing certificate authority. The certificatate is placed in the key ring.
+
+### Create a certificate authority and use it to self-sign a certificate
+
+The `zwe init security` command takes its input from the `zowe.setup.security` section in the `zowe.yaml` file. To help with file customization, there are five sections in the file.
 
 ## Extended key usage
```

Comment:
Important, but organized in a way users wont expect.
We should break down discussion into keystore & truststore.

* keystore
    * valid certificates
        * extended key usage
    * verification strict/nonstrict/disabled
    * domains / alternate names
    * expiry
    * client side requirements (how to not appear self-signed)